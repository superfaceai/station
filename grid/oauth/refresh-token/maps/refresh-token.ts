/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable jest/no-export */
import { SuperfaceTest } from '@superfaceai/testing';

// function beforeRecordingSave(recordings: RecordingDefinitions) {
//   let pageAccessToken: string | undefined;

//   recordings.forEach(recording => {
//     if (pageAccessToken) {
//       recording.path = recording.path.replace(
//         pageAccessToken,
//         PAGE_ACCESS_TOKEN_REDACTED_VALUE
//       );
//     }
//     if (recording.response) {
//       const response = recording.response as { access_token?: string };
//       if (response.access_token) {
//         pageAccessToken = response.access_token;
//         recording.response = {
//           ...response,
//           access_token: PAGE_ACCESS_TOKEN_REDACTED_VALUE,
//         };
//       }
//     }
//   });
// }

type UseCaseInput = {
  refreshToken?: string;
  clientId?: string;
  clientSecret?: string;
};

export const refreshTokenTest = (
  provider: string,
  input: UseCaseInput
): void => {
  describe(`oauth/refresh-token/${provider}`, () => {
    let superfaceRefreshToken: SuperfaceTest;

    beforeEach(() => {
      superfaceRefreshToken = new SuperfaceTest({
        profile: 'oauth/refresh-token',
        provider,
      });
    });

    // sanity check
    it('has all env values setup correctly', () => {
      expect(input).toEqual({
        refreshToken: expect.any(String),
        clientId: expect.any(String),
        clientSecret: expect.any(String),
      });
    });

    describe('GetAccessTokenFromRefreshToken', () => {
      describe('with all parameters in input', () => {
        it('issues an access token', async () => {
          const result = await superfaceRefreshToken.run({
            useCase: 'GetAccessTokenFromRefreshToken',
            input,
          });

          expect(result.isOk()).toBeTruthy();
          expect(result.unwrap()).toMatchSnapshot();
        });
      });

      describe('with integration parameters', () => {
        it('issues an access token', async () => {
          const { refreshToken } = input;
          const result = await superfaceRefreshToken.run({
            useCase: 'GetAccessTokenFromRefreshToken',
            input: { refreshToken },
          });

          expect(result.isOk()).toBeTruthy();
          expect(result.unwrap()).toMatchSnapshot();
        });
      });
    });
  });
};
