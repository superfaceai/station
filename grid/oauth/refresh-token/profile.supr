name = "oauth/refresh-token"
version = "1.0.0"

"""
Get Access Token from Refresh Token
Issue a new access token based on previously issued refresh token for OAuth 2.0 compatible providers.
Client authentication can be provided directly as input, or, if supported, set as provider parameter.
"""
usecase GetAccessTokenFromRefreshToken unsafe {
  input {
    """
    Refresh Token
    The refresh token previously issued to the client.
    """
    refreshToken! string!

    """
    Client ID
    Client authentication; may be also accepted as provider parameter.
    """
    clientId string!

    """
    Client Secret
    Client authentication; may be also accepted as provider parameter.
    """
    clientSecret string!
  }

  result {
    accessToken! string!
    expiresIn! number!
    scopes [string]
    tokenType string
  }

  error {
    status string
    message string
  }
}
