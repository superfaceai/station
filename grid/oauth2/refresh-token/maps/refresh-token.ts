/* eslint-disable jest/expect-expect */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable jest/no-export */
import {
  RecordingDefinitions,
  SuperfaceTest,
  TestingReturn,
} from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

type UseCaseInput = {
  refreshToken?: string;
  clientId?: string;
  clientSecret?: string;
};

type UseCaseResult = {
  accessToken: string;
  refreshToken?: string;
  [key: string]: unknown;
};

const ACCESS_TOKEN_REDACTED_VALUE = 'XXX_ACCESS_TOKEN_REDACTED_XXX';
export const REFRESH_TOKEN_REDACTED_VALUE = 'XXX_REFRESH_TOKEN_REDACTED_XXX';

function generateProcessingOptions(input: UseCaseInput) {
  const refreshToken = input.refreshToken || '';
  const refreshTokenEncoded = encodeURIComponent(refreshToken);

  function beforeRecordingLoad(recordings: RecordingDefinitions) {
    for (const recording of recordings) {
      if (refreshToken && typeof recording.body === 'string') {
        recording.body = recording.body.replace(
          REFRESH_TOKEN_REDACTED_VALUE,
          refreshToken
        );
      }
    }
  }

  function beforeRecordingSave(recordings: RecordingDefinitions) {
    for (const recording of recordings) {
      if (refreshToken && typeof recording.body === 'string') {
        recording.body = recording.body.replace(
          refreshToken,
          REFRESH_TOKEN_REDACTED_VALUE
        );
        recording.body = recording.body.replace(
          refreshTokenEncoded,
          REFRESH_TOKEN_REDACTED_VALUE
        );
      }
      // Remove tokens in standard body response
      if (recording.response) {
        const response = recording.response as UseCaseResult;
        const modifiedProps: Record<string, string> = {};
        if (response.access_token) {
          modifiedProps.access_token = ACCESS_TOKEN_REDACTED_VALUE;
        }
        if (response.refresh_token) {
          modifiedProps.refresh_token = REFRESH_TOKEN_REDACTED_VALUE;
        }

        recording.response = { ...response, ...modifiedProps };
      }
    }
  }

  return {
    beforeRecordingSave,
    beforeRecordingLoad,
  };
}

export const refreshTokenTest = (
  provider: string,
  input: UseCaseInput,
  onRefreshPerform?: (result: UseCaseResult) => void
): void => {
  describe(`oauth2/refresh-token/${provider}`, () => {
    let superfaceRefreshToken: SuperfaceTest;
    const currentInput = { ...input };

    function expectResult(result: TestingReturn) {
      expect(result.isOk()).toBe(true);
      const resultUnwrapped = result.unwrap() as {
        accessToken: string;
        refreshToken?: string;
      };
      onRefreshPerform?.(resultUnwrapped);
      const { refreshToken, ...rest } = resultUnwrapped;
      expect(rest).toMatchSnapshot({
        accessToken: expect.any(String),
      });
      if (refreshToken) {
        currentInput.refreshToken = refreshToken;
      }
    }

    beforeEach(() => {
      superfaceRefreshToken = buildSuperfaceTest({
        profile: 'oauth2/refresh-token',
        useCase: 'GetAccessTokenFromRefreshToken',
        provider,
      });
    });

    // sanity check
    it('has all env values setup correctly', () => {
      expect(currentInput).toEqual({
        refreshToken: expect.any(String),
        clientId: expect.any(String),
        clientSecret: expect.any(String),
      });
    });

    describe('GetAccessTokenFromRefreshToken', () => {
      describe('with all parameters in input', () => {
        it('issues an access token', async () => {
          const result = await superfaceRefreshToken.run(
            {
              input: currentInput,
            },
            generateProcessingOptions(currentInput)
          );
          expectResult(result);
        });
      });

      describe('with integration parameters', () => {
        it('issues an access token', async () => {
          const { refreshToken } = currentInput;
          const result = await superfaceRefreshToken.run(
            {
              input: { refreshToken },
            },
            generateProcessingOptions(currentInput)
          );
          expectResult(result);
        });
      });
    });
  });
};
