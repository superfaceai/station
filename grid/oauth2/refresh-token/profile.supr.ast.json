{
  "kind": "ProfileDocument",
  "header": {
    "kind": "ProfileHeader",
    "scope": "oauth2",
    "name": "refresh-token",
    "version": {
      "major": 1,
      "minor": 0,
      "patch": 0
    },
    "location": {
      "start": {
        "line": 1,
        "column": 1,
        "charIndex": 0
      },
      "end": {
        "line": 2,
        "column": 18,
        "charIndex": 47
      }
    }
  },
  "definitions": [
    {
      "kind": "UseCaseDefinition",
      "useCaseName": "GetAccessTokenFromRefreshToken",
      "safety": "unsafe",
      "input": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ObjectDefinition",
          "fields": [
            {
              "kind": "FieldDefinition",
              "fieldName": "refreshToken",
              "required": true,
              "type": {
                "kind": "NonNullDefinition",
                "type": {
                  "kind": "PrimitiveTypeName",
                  "name": "string",
                  "location": {
                    "start": {
                      "line": 15,
                      "column": 19,
                      "charIndex": 462
                    },
                    "end": {
                      "line": 15,
                      "column": 25,
                      "charIndex": 468
                    }
                  }
                },
                "location": {
                  "start": {
                    "line": 15,
                    "column": 19,
                    "charIndex": 462
                  },
                  "end": {
                    "line": 15,
                    "column": 26,
                    "charIndex": 469
                  }
                }
              },
              "location": {
                "start": {
                  "line": 15,
                  "column": 5,
                  "charIndex": 448
                },
                "end": {
                  "line": 15,
                  "column": 26,
                  "charIndex": 469
                }
              },
              "documentation": {
                "title": "Refresh Token",
                "description": "The refresh token previously issued to the client.",
                "location": {
                  "start": {
                    "line": 11,
                    "column": 5,
                    "charIndex": 359
                  },
                  "end": {
                    "line": 14,
                    "column": 8,
                    "charIndex": 443
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "clientId",
              "required": false,
              "type": {
                "kind": "NonNullDefinition",
                "type": {
                  "kind": "PrimitiveTypeName",
                  "name": "string",
                  "location": {
                    "start": {
                      "line": 21,
                      "column": 14,
                      "charIndex": 585
                    },
                    "end": {
                      "line": 21,
                      "column": 20,
                      "charIndex": 591
                    }
                  }
                },
                "location": {
                  "start": {
                    "line": 21,
                    "column": 14,
                    "charIndex": 585
                  },
                  "end": {
                    "line": 21,
                    "column": 21,
                    "charIndex": 592
                  }
                }
              },
              "location": {
                "start": {
                  "line": 21,
                  "column": 5,
                  "charIndex": 576
                },
                "end": {
                  "line": 21,
                  "column": 21,
                  "charIndex": 592
                }
              },
              "documentation": {
                "title": "Client ID",
                "description": "Client authentication; may be also accepted as provider parameter.",
                "location": {
                  "start": {
                    "line": 17,
                    "column": 5,
                    "charIndex": 475
                  },
                  "end": {
                    "line": 20,
                    "column": 8,
                    "charIndex": 571
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "clientSecret",
              "required": false,
              "type": {
                "kind": "NonNullDefinition",
                "type": {
                  "kind": "PrimitiveTypeName",
                  "name": "string",
                  "location": {
                    "start": {
                      "line": 27,
                      "column": 18,
                      "charIndex": 716
                    },
                    "end": {
                      "line": 27,
                      "column": 24,
                      "charIndex": 722
                    }
                  }
                },
                "location": {
                  "start": {
                    "line": 27,
                    "column": 18,
                    "charIndex": 716
                  },
                  "end": {
                    "line": 27,
                    "column": 25,
                    "charIndex": 723
                  }
                }
              },
              "location": {
                "start": {
                  "line": 27,
                  "column": 5,
                  "charIndex": 703
                },
                "end": {
                  "line": 27,
                  "column": 25,
                  "charIndex": 723
                }
              },
              "documentation": {
                "title": "Client Secret",
                "description": "Client authentication; may be also accepted as provider parameter.",
                "location": {
                  "start": {
                    "line": 23,
                    "column": 5,
                    "charIndex": 598
                  },
                  "end": {
                    "line": 26,
                    "column": 8,
                    "charIndex": 698
                  }
                }
              }
            }
          ],
          "location": {
            "start": {
              "line": 10,
              "column": 9,
              "charIndex": 353
            },
            "end": {
              "line": 28,
              "column": 4,
              "charIndex": 727
            }
          }
        },
        "location": {
          "start": {
            "line": 10,
            "column": 3,
            "charIndex": 347
          },
          "end": {
            "line": 28,
            "column": 4,
            "charIndex": 727
          }
        }
      },
      "result": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ObjectDefinition",
          "fields": [
            {
              "kind": "FieldDefinition",
              "fieldName": "accessToken",
              "required": true,
              "type": {
                "kind": "NonNullDefinition",
                "type": {
                  "kind": "PrimitiveTypeName",
                  "name": "string",
                  "location": {
                    "start": {
                      "line": 35,
                      "column": 18,
                      "charIndex": 820
                    },
                    "end": {
                      "line": 35,
                      "column": 24,
                      "charIndex": 826
                    }
                  }
                },
                "location": {
                  "start": {
                    "line": 35,
                    "column": 18,
                    "charIndex": 820
                  },
                  "end": {
                    "line": 35,
                    "column": 25,
                    "charIndex": 827
                  }
                }
              },
              "location": {
                "start": {
                  "line": 35,
                  "column": 5,
                  "charIndex": 807
                },
                "end": {
                  "line": 35,
                  "column": 25,
                  "charIndex": 827
                }
              },
              "documentation": {
                "title": "Access Token",
                "description": "Newly issued access token",
                "location": {
                  "start": {
                    "line": 31,
                    "column": 5,
                    "charIndex": 744
                  },
                  "end": {
                    "line": 34,
                    "column": 8,
                    "charIndex": 802
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "expiresIn",
              "required": false,
              "type": {
                "kind": "NonNullDefinition",
                "type": {
                  "kind": "PrimitiveTypeName",
                  "name": "number",
                  "location": {
                    "start": {
                      "line": 40,
                      "column": 15,
                      "charIndex": 937
                    },
                    "end": {
                      "line": 40,
                      "column": 21,
                      "charIndex": 943
                    }
                  }
                },
                "location": {
                  "start": {
                    "line": 40,
                    "column": 15,
                    "charIndex": 937
                  },
                  "end": {
                    "line": 40,
                    "column": 22,
                    "charIndex": 944
                  }
                }
              },
              "location": {
                "start": {
                  "line": 40,
                  "column": 5,
                  "charIndex": 927
                },
                "end": {
                  "line": 40,
                  "column": 22,
                  "charIndex": 944
                }
              },
              "documentation": {
                "title": "Expires in (seconds)",
                "description": "Duration of time the access token is granted for.",
                "location": {
                  "start": {
                    "line": 36,
                    "column": 5,
                    "charIndex": 832
                  },
                  "end": {
                    "line": 39,
                    "column": 8,
                    "charIndex": 922
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "scopes",
              "required": false,
              "type": {
                "kind": "ListDefinition",
                "elementType": {
                  "kind": "PrimitiveTypeName",
                  "name": "string",
                  "location": {
                    "start": {
                      "line": 45,
                      "column": 13,
                      "charIndex": 1035
                    },
                    "end": {
                      "line": 45,
                      "column": 19,
                      "charIndex": 1041
                    }
                  }
                },
                "location": {
                  "start": {
                    "line": 45,
                    "column": 12,
                    "charIndex": 1034
                  },
                  "end": {
                    "line": 45,
                    "column": 20,
                    "charIndex": 1042
                  }
                }
              },
              "location": {
                "start": {
                  "line": 45,
                  "column": 5,
                  "charIndex": 1027
                },
                "end": {
                  "line": 45,
                  "column": 20,
                  "charIndex": 1042
                }
              },
              "documentation": {
                "title": "Scopes",
                "description": "Authorized scopes for the issued access token.",
                "location": {
                  "start": {
                    "line": 41,
                    "column": 5,
                    "charIndex": 949
                  },
                  "end": {
                    "line": 44,
                    "column": 8,
                    "charIndex": 1022
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "tokenType",
              "required": false,
              "type": {
                "kind": "PrimitiveTypeName",
                "name": "string",
                "location": {
                  "start": {
                    "line": 50,
                    "column": 15,
                    "charIndex": 1137
                  },
                  "end": {
                    "line": 50,
                    "column": 21,
                    "charIndex": 1143
                  }
                }
              },
              "location": {
                "start": {
                  "line": 50,
                  "column": 5,
                  "charIndex": 1127
                },
                "end": {
                  "line": 50,
                  "column": 21,
                  "charIndex": 1143
                }
              },
              "documentation": {
                "title": "Token type",
                "description": "The type of token this is, usually \"Bearer\".",
                "location": {
                  "start": {
                    "line": 46,
                    "column": 5,
                    "charIndex": 1047
                  },
                  "end": {
                    "line": 49,
                    "column": 8,
                    "charIndex": 1122
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "refreshToken",
              "required": false,
              "type": {
                "kind": "PrimitiveTypeName",
                "name": "string",
                "location": {
                  "start": {
                    "line": 56,
                    "column": 18,
                    "charIndex": 1287
                  },
                  "end": {
                    "line": 56,
                    "column": 24,
                    "charIndex": 1293
                  }
                }
              },
              "location": {
                "start": {
                  "line": 56,
                  "column": 5,
                  "charIndex": 1274
                },
                "end": {
                  "line": 56,
                  "column": 24,
                  "charIndex": 1293
                }
              },
              "documentation": {
                "title": "Refresh token",
                "description": "Provider may issue a new refresh token, invalidating the previously used refresh token",
                "location": {
                  "start": {
                    "line": 52,
                    "column": 5,
                    "charIndex": 1149
                  },
                  "end": {
                    "line": 55,
                    "column": 8,
                    "charIndex": 1269
                  }
                }
              }
            }
          ],
          "location": {
            "start": {
              "line": 30,
              "column": 10,
              "charIndex": 738
            },
            "end": {
              "line": 57,
              "column": 4,
              "charIndex": 1297
            }
          }
        },
        "location": {
          "start": {
            "line": 30,
            "column": 3,
            "charIndex": 731
          },
          "end": {
            "line": 57,
            "column": 4,
            "charIndex": 1297
          }
        }
      },
      "error": {
        "kind": "UseCaseSlotDefinition",
        "value": {
          "kind": "ObjectDefinition",
          "fields": [
            {
              "kind": "FieldDefinition",
              "fieldName": "error",
              "required": false,
              "type": {
                "kind": "EnumDefinition",
                "values": [
                  {
                    "kind": "EnumValue",
                    "name": "invalid_request",
                    "value": "invalid_request",
                    "location": {
                      "start": {
                        "line": 62,
                        "column": 7,
                        "charIndex": 1349
                      },
                      "end": {
                        "line": 62,
                        "column": 22,
                        "charIndex": 1364
                      }
                    }
                  },
                  {
                    "kind": "EnumValue",
                    "name": "invalid_client",
                    "value": "invalid_client",
                    "location": {
                      "start": {
                        "line": 63,
                        "column": 7,
                        "charIndex": 1371
                      },
                      "end": {
                        "line": 63,
                        "column": 21,
                        "charIndex": 1385
                      }
                    }
                  },
                  {
                    "kind": "EnumValue",
                    "name": "invalid_grant",
                    "value": "invalid_grant",
                    "location": {
                      "start": {
                        "line": 64,
                        "column": 7,
                        "charIndex": 1392
                      },
                      "end": {
                        "line": 64,
                        "column": 20,
                        "charIndex": 1405
                      }
                    }
                  },
                  {
                    "kind": "EnumValue",
                    "name": "invalid_scope",
                    "value": "invalid_scope",
                    "location": {
                      "start": {
                        "line": 65,
                        "column": 7,
                        "charIndex": 1412
                      },
                      "end": {
                        "line": 65,
                        "column": 20,
                        "charIndex": 1425
                      }
                    }
                  },
                  {
                    "kind": "EnumValue",
                    "name": "unauthorized_client",
                    "value": "unauthorized_client",
                    "location": {
                      "start": {
                        "line": 66,
                        "column": 7,
                        "charIndex": 1432
                      },
                      "end": {
                        "line": 66,
                        "column": 26,
                        "charIndex": 1451
                      }
                    }
                  },
                  {
                    "kind": "EnumValue",
                    "name": "unsupported_grant_type",
                    "value": "unsupported_grant_type",
                    "location": {
                      "start": {
                        "line": 67,
                        "column": 7,
                        "charIndex": 1458
                      },
                      "end": {
                        "line": 67,
                        "column": 29,
                        "charIndex": 1480
                      }
                    }
                  },
                  {
                    "kind": "EnumValue",
                    "name": "unknown_response",
                    "value": "unknown_response",
                    "location": {
                      "start": {
                        "line": 69,
                        "column": 7,
                        "charIndex": 1529
                      },
                      "end": {
                        "line": 69,
                        "column": 23,
                        "charIndex": 1545
                      }
                    }
                  }
                ],
                "location": {
                  "start": {
                    "line": 61,
                    "column": 11,
                    "charIndex": 1336
                  },
                  "end": {
                    "line": 70,
                    "column": 6,
                    "charIndex": 1551
                  }
                }
              },
              "location": {
                "start": {
                  "line": 61,
                  "column": 5,
                  "charIndex": 1330
                },
                "end": {
                  "line": 70,
                  "column": 6,
                  "charIndex": 1551
                }
              },
              "documentation": {
                "title": "Error code",
                "location": {
                  "start": {
                    "line": 60,
                    "column": 5,
                    "charIndex": 1313
                  },
                  "end": {
                    "line": 60,
                    "column": 17,
                    "charIndex": 1325
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "description",
              "required": false,
              "location": {
                "start": {
                  "line": 72,
                  "column": 5,
                  "charIndex": 1580
                },
                "end": {
                  "line": 72,
                  "column": 16,
                  "charIndex": 1591
                }
              },
              "documentation": {
                "title": "Error description",
                "location": {
                  "start": {
                    "line": 71,
                    "column": 5,
                    "charIndex": 1556
                  },
                  "end": {
                    "line": 71,
                    "column": 24,
                    "charIndex": 1575
                  }
                }
              }
            },
            {
              "kind": "FieldDefinition",
              "fieldName": "link",
              "required": false,
              "location": {
                "start": {
                  "line": 74,
                  "column": 5,
                  "charIndex": 1625
                },
                "end": {
                  "line": 74,
                  "column": 9,
                  "charIndex": 1629
                }
              },
              "documentation": {
                "title": "Link with error detail",
                "location": {
                  "start": {
                    "line": 73,
                    "column": 5,
                    "charIndex": 1596
                  },
                  "end": {
                    "line": 73,
                    "column": 29,
                    "charIndex": 1620
                  }
                }
              }
            }
          ],
          "location": {
            "start": {
              "line": 59,
              "column": 9,
              "charIndex": 1307
            },
            "end": {
              "line": 75,
              "column": 4,
              "charIndex": 1633
            }
          }
        },
        "location": {
          "start": {
            "line": 59,
            "column": 3,
            "charIndex": 1301
          },
          "end": {
            "line": 75,
            "column": 4,
            "charIndex": 1633
          }
        }
      },
      "location": {
        "start": {
          "line": 9,
          "column": 1,
          "charIndex": 297
        },
        "end": {
          "line": 76,
          "column": 2,
          "charIndex": 1635
        }
      },
      "documentation": {
        "title": "Get Access Token from Refresh Token",
        "description": "Issue a new access token based on previously issued refresh token for OAuth 2.0 compatible providers.\nClient authentication can be provided directly as input, or, if supported, set as provider parameter.",
        "location": {
          "start": {
            "line": 4,
            "column": 1,
            "charIndex": 49
          },
          "end": {
            "line": 8,
            "column": 4,
            "charIndex": 296
          }
        }
      }
    }
  ],
  "location": {
    "start": {
      "line": 1,
      "column": 1,
      "charIndex": 0
    },
    "end": {
      "line": 76,
      "column": 2,
      "charIndex": 1635
    }
  },
  "astMetadata": {
    "astVersion": {
      "major": 1,
      "minor": 2,
      "patch": 0
    },
    "parserVersion": {
      "major": 1,
      "minor": 2,
      "patch": 0
    },
    "sourceChecksum": "54ebd09eaf7e588ce15ffffeb238084b77e497eff4f01d4860b0b4de307975df"
  }
}
