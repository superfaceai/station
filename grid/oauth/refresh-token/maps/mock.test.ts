import { refreshTokenTest } from './refresh-token';

const input = {
  refreshToken: 'foobar',
  clientId: 'clientId',
  clientSecret: 'clientSecret',
};
refreshTokenTest('mock', input);
