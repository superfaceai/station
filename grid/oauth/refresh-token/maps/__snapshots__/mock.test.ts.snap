// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`oauth/refresh-token/mock GetAccessTokenFromRefreshToken with all parameters in input issues an access token 1`] = `
Object {
  "expiresIn": 3600,
  "scopes": Array [
    "read",
    "write",
  ],
  "tokenType": "Bearer",
}
`;

exports[`oauth/refresh-token/mock GetAccessTokenFromRefreshToken with integration parameters issues an access token 1`] = `
Object {
  "expiresIn": 3600,
  "scopes": Array [
    "read",
    "write",
  ],
  "tokenType": "Bearer",
}
`;
